examples/rcn.py

```diff
@@ -0,0 +1,390 @@
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.4
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %%
+# %matplotlib inline
+import os
+import time
+
+import jax
+import matplotlib.pyplot as plt
+import numpy as np
+from jax import numpy as jnp
+from jax import tree_util
+from scipy.ndimage import maximum_filter
+from scipy.signal import fftconvolve
+from sklearn.datasets import fetch_openml
+
+from pgmax.fg import graph, groups
+
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+
+# %% [markdown]
+# # 1. Load the data
+# %%
+hps, vps = 12, 12
+
+# Use train_size = 100 if you have a gpu with atleast 8Gbs memory.
+# Recommend that jax is installed with cuda enabled for this option.
+train_size = 100
+test_size = 20
+
+
+def fetch_mnist_dataset(test_size: int, seed: int = 5):
+    """Returns test images sampled randomly from the set of MNIST images.
+    Args:
+        test_size: Desired number of test images.
+    Returns:
+        test_set: A list of length test_size containing images from the MNIST test dataset.
+        test_labels: Corresponding labels for the test images.
+    """
+
+    mnist_train_size = 60000
+    num_per_class = test_size // 10
+
+    print("Fetching the MNIST dataset")
+    dataset = fetch_openml("mnist_784", as_frame=False, cache=True)
+    print("Successfully downloaded the MNIST dataset")
+
+    mnist_images = dataset["data"]
+    mnist_labels = dataset["target"].astype("int")
+    full_mnist_test_images = mnist_images[mnist_train_size:]
+    full_mnist_test_labels = mnist_labels[mnist_train_size:]
+
+    np.random.seed(seed)
+
+    test_set = []
+    test_labels = []
+    for i in range(10):
+        idxs = np.random.choice(
+            np.argwhere(full_mnist_test_labels == i)[:, 0], num_per_class
+        )
+        for idx in idxs:
+            img = full_mnist_test_images[idx].reshape(28, 28)
+            img_arr = jax.image.resize(image=img, shape=(112, 112), method="bicubic")
+            img = jnp.pad(
+                img_arr,
+                pad_width=tuple([(p, p) for p in (44, 44)]),
+                mode="constant",
+                constant_values=0,
+            )
+
+            test_set.append(img)
+            test_labels.append(i)
+
+    return test_set, np.array(test_labels)
+
+
+# %%
+test_set, test_labels = fetch_mnist_dataset(test_size)
+train_labels = (
+    np.array([[i] * (train_size // 10) for i in range(10)]).reshape(1, -1).squeeze()
+)
+
+# %% [markdown]
+# # 2. Load the model
+
+# %%
+data = np.load("example_data/rcn_100.npz", allow_pickle=True, encoding="latin1")
+frcs, edges, suppression_masks, filters = (
+    data["frcs"][:train_size],
+    data["edges"][:train_size],
+    data["suppression_masks"],
+    data["filters"],
+)
+
+M = (2 * hps + 1) * (2 * vps + 1)
+
+# %% [markdown]
+# # 3. Visualize loaded model
```

Comment:
For visualizing the model, please also visualize the corresponding training image side by side